crates/symbolicator/src/utils/sentry.rs

```diff
@@ -8,3 +15,65 @@ pub trait ConfigureScope {
         sentry::configure_scope(|scope| self.to_scope(scope));
     }
 }
+
+#[derive(Clone)]
+pub struct SentryRequestLayer;
+
+#[derive(Clone)]
+pub struct SentryRequestService<S> {
+    service: S,
+}
+
+impl<S> Layer<S> for SentryRequestLayer {
+    type Service = SentryRequestService<S>;
+
+    fn layer(&self, service: S) -> Self::Service {
+        Self::Service { service }
+    }
+}
+
+impl<S> Service<Request<Body>> for SentryRequestService<S>
+where
+    S: Service<Request<Body>>,
+{
+    type Response = S::Response;
+    type Error = S::Error;
+    type Future = S::Future;
+
+    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        self.service.poll_ready(cx)
+    }
+
+    fn call(&mut self, request: Request<Body>) -> Self::Future {
+        sentry::configure_scope(|scope| {
```

Comment:
Shouldn't we turn the future returned from `self.service.call()` into a `SentryFuture` and use `with_scope` instead so that this is all popped back off the scope when this is done?

Reply:
The `sentry-tower` layer makes sure that every request gets its own fresh hub and scope, and if I understand the tower concepts correctly, all child layers/routes get the correct hub.